README.md

```diff
@@ -1,21 +1,97 @@
+# Compile instruction
+
+## Before compiling
+
+please check the [Releases](https://github.com/Franklalalala/usenauty/releases). There are pre-compiled linux and windows versions, respectively.
+
+## Compile a linux version
+
+The [CXX standard](https://en.wikipedia.org/wiki/C%2B%2B17) is set to be **17**, which means the gcc version need to be 8 or higher. Additionally, the cmake version needs to be **3.1** or higher.
+
+With these conditions, one can follow these steps:
+
+1. Load gcc and cmake modules. For example:
+
+```
+module load compiler/gcc/8.4.0
+module load compiler/cmake/3.15.6
+```
+
+2. Make sure the compiler works.
+
+```
+which gcc
+# /public/software/compiler/gnu/gcc-8.4.0/bin/gcc # The gcc path
+export  CC=/public/software/compiler/gnu/gcc-8.4.0/bin/gcc
+export  CXX=/public/software/compiler/gnu/gcc-8.4.0/bin/g++
```

Comment:
In my view, if the module set, these environment variables are unnecessary to be set again, unless one have system issue on their own.